### PR TITLE
Prepare Castro for upcoming changes in AMReX

### PR DESCRIPTION
**Pull request by WeiqunZhang**
There are some upcoming changes in AMReX's Geometry class. This pull-request makes Castro work with the current AMReX development branch and the future version.  I have tested this in CCSE's regression test suite.

### CODE REVIEW

**Comment by zingale** [comment]
I'll run out suite and then merge

**Comment by WeiqunZhang** [comment]
I updated `Util/ConvertCheckpoint`, but I didn't test it.  However, it seems that it has not been used for a long time.  It didn't actually compile before my change.  It compiles now.

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -753,12 +755,12 @@ Castro::initMFs()
 	mass_fluxes[dir].reset(new MultiFab(get_new_data(State_Type).boxArray(), dmap, 1, 0));
 
 #if (BL_SPACEDIM <= 2)
-    if (!Geometry::IsCartesian())
+    if (!Geom().IsCartesian())
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp]: What's the difference between DefaultGeometry() and Geom()?

Comment (WeiqunZhang) [reply Source/driver/Castro.cpp]: `Geom()` is a member function of `AmrLevel`, whereas `DefaultGeometry` is a free function in amrex namespace.  For `IsCartesian()`, both are fine.  But for things like `CellSize`, one has to use `Geom()` or `Geom(int level)` in `Amr` class.

**Pull request merged**